src/profile-logic/process-profile.js

```diff
@@ -28,59 +55,60 @@ import { getEmptyTaskTracerData } from './task-tracer';
  * drastically reduces the number of JS objects the JS engine has to deal with,
  * resulting in fewer GC pauses and hopefully better performance.
  *
- * @param {object} geckoTable A data table of the form `{ schema, data }`.
- * @return {object} A data table of the form `{ length: number, field1: array, field2: array }`
+ * e.g Take geckoTable A data table of the form
+ *   `{ schema, data }`.
+ * And turn it into a data table of the form
+ *  `{ length: number, field1: array, field2: array }`
  */
-function toStructOfArrays(geckoTable) {
+function _toStructOfArrays(geckoTable: Object): Object {
```

Comment:
This thing was really tough to get typed. I feel like it's good enough for now, even if it's not perfect for coverage.